mcs/class/System/System.Security.Cryptography.X509Certificates/X509Certificate2.cs

```diff
@@ -67,8 +67,15 @@ public X509Certificate2 ()
 		}
 
 		public X509Certificate2 (byte[] rawData)
-			: base (rawData)
 		{
+			// MONO: temporary hack until `X509CertificateImplApple` derives from
```

Comment:
This is `X509Certificate2`, so we need an `X509Certificate2Impl` implementation.  At the moment, this will never be a native "apple" certificate, but the legacy fallback that's using `Mono.Security.X509`.  Eventually, the native apple implementation will derive from `X509Certificate2Impl`, but we aren't there yet.

In the base class, we want to use the native apple implementation whenever possible (and we have `SecCertificate` interoperability depending on this functionality), so the `X509Certificate(byte[])` constructor does that by calling `X509Helper.Import(byte[])`.

However, this also means that by calling the base constructor, we might get an `X509CertificateImpl` implementation instead of an `X509Certificate2Impl`.  This is why we temporarily need this mono-specific version.

Does this answer your question>